File: backend/commons/security/security-entity/src/main/java/com/kraken/security/entity/owner/PublicOwner.java
package com.kraken.security.entity.owner;

import com.fasterxml.jackson.annotation.JsonCreator;
import lombok.Builder;
import lombok.Value;

@Value
public class PublicOwner implements Owner {

  public final static PublicOwner INSTANCE = new PublicOwner();

  @JsonCreator
  PublicOwner() {
    super();
  }

  @Override
  public OwnerType getType() {
    return OwnerType.PUBLIC;
  }
}

File: frontend/projects/security/src/lib/entities/public-owner.ts
import {Owner} from 'projects/security/src/lib/entities/owner';
import {OwnerType} from 'projects/security/src/lib/entities/owner-type';

export class PublicOwner implements Owner {
  constructor(
    public type: OwnerType = 'PUBLIC') {
  }
}
